sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java

```diff
@@ -140,6 +149,66 @@ public void testNullSubscription() {
     assertNotNull(DisplayData.from(read));
   }
 
+  /**
+   * {@link PipelineOptions} with a ValueProvider.
+   */
+  public interface RuntimeSubscriptionOptions extends PipelineOptions {
+    ValueProvider<String> getValue();
+    void setValue(ValueProvider<String> provider);
+  }
+
+  @Test
+  public void testValueProviderSubscription() {
+    RuntimeSubscriptionOptions options =
+        PipelineOptionsFactory.create().as(RuntimeSubscriptionOptions.class);
+    options.setValue(StaticValueProvider.of("projects/project/subscriptions/subscription"));
+    ValueProvider<String> subscription = options.getValue();
+    Read<String> pubsubRead = PubsubIO.readStrings().fromSubscription(subscription);
```

Comment:
You could simply pass a StaticValueProvider here

Reply:
Done.